lib/collection/src/lib.rs

```diff
@@ -238,15 +239,20 @@ impl Collection {
         operation: CollectionUpdateOperations,
         shard_selection: ShardId,
         wait: bool,
+        ip_to_address: &HashMap<u64, Uri>,
     ) -> CollectionResult<UpdateResult> {
         let local_shard = self.local_shard_by_id(shard_selection)?;
-        local_shard.get().update(operation.clone(), wait).await
+        local_shard
+            .get()
+            .update(operation.clone(), wait, ip_to_address)
+            .await
     }
 
     pub async fn update_from_client(
         &self,
         operation: CollectionUpdateOperations,
         wait: bool,
+        ip_to_address: &HashMap<u64, Uri>,
```

Comment:
I feel like the `ip_to_address` should be a part of `self`, otherwise it's a leak of abstraction. As a developer I should not care about cluster state if I want to make an update.

Reply:
We discussed this choice as well during the design phase with @e-ivkov .

The source of truth is the `RaftState` on the toc:

```rust
raft_state: Arc<std::sync::Mutex<PersistentRaftState>>,
```

exposed via 

```rust
pub fn peer_address_by_id(&self) -> Result<PeerAddressById, StorageError> {
     Ok(self.raft_state.lock()?.peer_address_by_id().clone())
}
```

Given that the `RemoteShard` in the collection module does not see any of the `RaftState` types in the storage, we have decided to go with a more explicit approach propagating only the HashMap at the call site.

What would be you preferred way to encapsulate that state safely in the `RemoteShard` struct or somewhere else?

Reply:
Alright I will give it a try :+1:

Reply:
In a future PR, we could introduce Egor's idea regarding having a read-only lock to prevent the remote shards from modifying the mappings.
I decided to keep it out for now for simplicity.